name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/views/SimpleMovingAverage.java

```diff
@@ -0,0 +1,84 @@
+package name.abuchen.portfolio.ui.views;
+
+import java.time.LocalDate;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import name.abuchen.portfolio.model.Security;
+import name.abuchen.portfolio.model.SecurityPrice;
+import name.abuchen.portfolio.money.Values;
+import name.abuchen.portfolio.ui.util.chart.TimelineChart;
+
+public class SimpleMovingAverage
+{
+    /**
+     * Calculates the Simple Moving Average for the given range of days from the given startDate on
+     * The method returns an object containing the X and Y Axes of the generated SMA
+     */
+    public static ChartLineSeriesAxes getSMA(Integer RangeSMA, Security security, LocalDate startDate)
+    {
+        int index;
+        LocalDate[] dates;
+        double[] values;
+        double sum = 0;
+        ChartLineSeriesAxes lineSeries = new ChartLineSeriesAxes();
+
+        List<SecurityPrice> prices = security.getPricesIncludingLatest();
+
+        if (startDate == null)
+        {
+            index = RangeSMA;
+            dates = new LocalDate[prices.size() - RangeSMA];
+            values = new double[prices.size() - RangeSMA];
+        }
+        else
+        {
+            index = Math.abs(Collections.binarySearch(prices, new SecurityPrice(startDate, 0),
+                            new SecurityPrice.ByDate()));
+
+            if (index >= prices.size())
+            {
+                // no data available
+                return null;
+            }
+            
+            if (index < RangeSMA) {
+                index = RangeSMA;
+            }
+            
+            dates = new LocalDate[prices.size() - index];
+            values = new double[prices.size() - index];
+        }
+
+        for (int ii = 0; index < prices.size(); index++, ii++)
+        {
+            List<SecurityPrice> filteredPrices = null;
+            int start = index - RangeSMA;
+            sum = 0;
+            if (start >= 0)
+            {
+                LocalDate indexDate = prices.get(index).getTime();
+                LocalDate indexDateMinusSMA = indexDate.minusDays(RangeSMA);
+                filteredPrices = prices.stream()
+                                .filter(p -> p.getTime().isAfter(indexDateMinusSMA) && p.getTime().isBefore(indexDate))
```

Comment:
Ich frage mich, ob mit der Bedingung ```p.getTime().isBefore(indexDate)``` der Kurs von dem gerade aktuellem Datum (also index) eben *nicht* in den SMA einfliesst.

Müsste man nicht lieber !indexDatex.isBefore(p.getTime()) sagen?